disperser/encoder/encoder_config.go

```diff
@@ -0,0 +1,197 @@
+package encoder
+
+import (
+	"fmt"
+
+	"github.com/Layr-Labs/eigenda/common"
+	"github.com/Layr-Labs/eigenda/common/aws"
+	"github.com/Layr-Labs/eigenda/common/config"
+	"github.com/Layr-Labs/eigenda/disperser/common/blobstore"
+	"github.com/Layr-Labs/eigenda/encoding"
+	"github.com/Layr-Labs/eigenda/encoding/v1/kzg"
+	"github.com/Layr-Labs/eigenda/relay/chunkstore"
+)
+
+var _ config.DocumentedConfig = (*RootEncoderConfig)(nil)
+
+// The root configuration for the encoder service. This config should be discarded after parsing
+// and only the sub-configs should be used. This is a safety mechanism to make it harder to
+// accidentally print/log the secret config.
+type RootEncoderConfig struct {
+	Config *EncoderConfig
+	Secret *EncoderSecretConfig
+}
+
+var _ config.VerifiableConfig = (*EncoderConfig)(nil)
+
+// Configuration for the encoder.
+type EncoderConfig struct {
+	// Encoder version (1 or 2)
+	EncoderVersion uint `docs:"required"`
+
+	// Port at which encoder listens for gRPC calls
+	GrpcPort string `docs:"required"`
```

Comment:
I don't think that `EncoderVersion` and `GrpcPort` should be marked as `required`. 

As I discussed with @cody-littley below: `required` can be understood to mean "the user must set this value, since there is no sensible default to be chosen". 

Since these both have sensible default values, they shouldn't be marked as `required`.